### PR TITLE
[functions] add windowFunnel function

### LINKED ISSUE #966: [functions] window funnel function
**Summary**

ClickHouse windowfunnel:
https://clickhouse.tech/docs/en/sql-reference/aggregate-functions/parametric-functions/#windowfunnel

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://datafuse.rs/policies/cla/

## Summary

[functions] add windowFunnel function

## Changelog

- New Feature

## Related Issues

Fixes #966

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: website/datafuse/mkdocs.yml [hunk 1]
```diff
@@ -164,6 +164,7 @@ nav:
           - minIf: sqlstatement/aggregate-functions/aggregate-min-if.md
           - maxIf: sqlstatement/aggregate-functions/aggregate-max-if.md
           - sumIf: sqlstatement/aggregate-functions/aggregate-sum-if.md
+          - windowFunnel: sqlstatement/aggregate-functions/aggregate-windowfunnel.md
```
Comment (BohuTANG) [inline website/datafuse/mkdocs.yml:167]: Cool.
Seems the `aggregate-windowfunnel.md` file is not attach in the patch.

**Comment by sundy-li** [comment]
Flaky test again:
https://github.com/datafuselabs/datafuse/pull/1519/checks?check_run_id=3381610130#step:5:1599

File: common/functions/src/aggregates/aggregate_avg.rs [hunk 2]
```diff
@@ -231,6 +231,7 @@ macro_rules! creator {
 
 pub fn try_create_aggregate_avg_function(
     display_name: &str,
+    _params: Vec<DataValue>,
```
Comment (zhang2014) [inline common/functions/src/aggregates/aggregate_avg.rs:234]: params are constant?

**Review by zhang2014 — APPROVED** [review]
LGTM

**Pull request merged**